gramps/gen/filters/rules/person/_hasrelationship.py

```diff
@@ -57,31 +66,28 @@ def __init__(self, arg, use_regex=False, use_case=False):
         super().__init__(arg, use_regex, use_case)
         self.relationship_type = None
 
-    def prepare(self, db, user):
+    def prepare(self, db: Database, user):
         """
         Prepare the rule. Things we only want to do once.
         """
         if self.list[1]:
             self.relationship_type = FamilyRelType()
             self.relationship_type.set_from_xml_str(self.list[1])
 
-    def apply(self, db, obj):
+    def apply_to_one(self, db: Database, obj: Person) -> bool:
         """
         Apply the rule. Return True on a match.
         """
         relationship_type = 0
         total_children = 0
-        number_relations = len(obj.get_family_handle_list())
+        number_relations = len(obj.family_list)
 
         # count children and look for a relationship type match
-        for handle in obj.get_family_handle_list():
+        for handle in obj.family_list:
             family = db.get_family_from_handle(handle)
             if family:
-                total_children += len(family.get_child_ref_list())
-                if (
-                    self.relationship_type
-                    and self.relationship_type == family.get_relationship()
-                ):
+                total_children += len(family.child_ref_list)
+                if self.relationship_type and self.relationship_type == family.type:
```

Comment:
```suggestion
                if self.relationship_type and (self.relationship_type == family.type):
```
Does using parenthesis to be explicit about precedence, rather than rely on precedence rules, make it easier for others in future?

Reply:
Yes, I think so.